src/tophat.h

```diff
@@ -35,6 +35,25 @@ typedef int32_t iu;
 
 #define LEN(a) (sizeof(a) / sizeof((a)[0]))
 #define MAX(a, b) ((a) > (b) ? (a) : (b))
+#define TH_ASSERT(err, e) \
+	do { \
+		if (!(e)) \
+			return err; \
```

Comment:
Is there a need for that? I think it shouldn't matter in case of a return.

Reply:
I think it would only matter if you put a semicolon there: `TH_ASSERT(0; blahblah`, but otherwise if this is not significant you can close this thread. I was just commenting.